rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java

```diff
@@ -330,6 +327,39 @@ private void addTracingKeys(NodeMsgTraceInfo nodeMsgTraceInfo) {
         }
     }
 
+    private void logStart(MessageTask task) {
+        MessageType messageType = task.getMessage().getMessageType();
+
+        if (task.getNodeMsgTraceInfo() == null) {
+            logger.trace("Start {} message task", messageType);
+            return;
+        }
+
+        long taskWaitTimeInSeconds = task.getNodeMsgTraceInfo().getLifeTimeInSeconds();
+        logger.trace("Start {} message task after [{}]s", messageType, taskWaitTimeInSeconds);
+
+        if (taskWaitTimeInSeconds >= QUEUED_TIME_TO_WARN_LIMIT) {
+            logger.debug("Task {} was waiting too much in the queue: [{}]s", messageType, taskWaitTimeInSeconds);
+            recentIdleTime = true;
+        }
+    }
+
+    @VisibleForTesting
+    void logEnd(MessageTask task, long startNanos) {
+        Duration processTime = Duration.ofNanos(System.nanoTime() - startNanos);
+
+        Message message = task.getMessage();
+        boolean isBlockRelated = message.getMessageType() == MessageType.BLOCK_MESSAGE || message.getMessageType() == MessageType.BODY_RESPONSE_MESSAGE;
+        boolean isBlockSlow = isBlockRelated && BlockUtils.tooMuchProcessTime(processTime.toNanos());
+        boolean isOtherSlow = !isBlockRelated && processTime.getSeconds() > PROCESSING_TIME_TO_WARN_LIMIT;
+
+        if (isBlockSlow || isOtherSlow) {
```

Comment:
i believe you'd like to be warned if block processing (`isBlockRelated == true`) is slow, vs others? that being the case might be worth splitting the logic of the warning..

Reply:
We will know it is block related as message type is printed in the log, check `Message[{}]` part, so we will be able to configure individual alerts per message type.